Wrap up

n-osborne opened the pull request:
This PR hopefully brings the last changes to be able to merge into main.

Mainly:
- generate the last pieces of the stm module
- some last minutes fixes
- new tests
- dependencies

shym commented on plugins/qcheck-stm/test/expect_stm.expected:
```diff
@@ -92,7 +94,10 @@ module Spec =
       | To_list -> Res ((list char), (to_list sut__010_))
   end
 module STMTests = (STM_sequential.Make)(Spec)
+<<<<<<< HEAD
```
This should not appear.

shym commented on plugins/qcheck-stm/src/stm_of_ir.ml:
```diff
@@ -576,6 +576,7 @@ let stm config ir =
   let spec_expr =
     pmod_structure
       [
+        open_mod "STM";
```
The diff of that commit does not match the message, I imagine there was some mixed up somewhere.

shym commented on plugins/qcheck-stm/test/expect_stm.expected:
```diff
@@ -1,6 +1,4 @@
 open Lib
-open QCheck
```
The changes of that commit (dbab08) do not match the commit message.

shym commented on ortac-qcheck-stm.opam:
```diff
@@ -30,6 +30,9 @@ depends: [
   "ppxlib" {>= "0.26.0"}
   "gospel"
   "ortac"
+  "ortac-runtime" {with-test}
```
I have a little doubt there: indeed we can install our plugin and even use it without those dependencies, but the plugin is then clearly useless. As we are generating an executable command, I think putting all the dependencies that are required to actually fully use it is rather fair. (There debian would draw its nuances of recommends and suggests but I don’t think we have that).

shym reviewed (commented):
I see two commits whose messages do not match the content. I think there might have been a mishap with commit 557ad08 containing a lot of code it was not meant to contain, maybe the content of those commits. `git reflog` might come to the rescue to dig up what went wrong during some rebase?

n-osborne committed:
Fix `postcond` pattern matching

`Res` is an extensible type, so pattern matching needs a catch-all.

n-osborne committed:
Fix argument order in cli

Argument order in cli should be the same than for the function call.
Also, expression coming before the type is more natural.

n-osborne committed:
Wrap up the stm module

- add cleanup function
- add sut type declaration
- organize content in module, open needed modules and instantiate functor
- delcare and call tests

n-osborne committed:
Update tests

Use `ortac` cli to generate the stm file

n-osborne committed:
Update dependencies

n-osborne committed:
Add Samuel as an author

n-osborne commented:
History is rewritten and should be correct :crossed_fingers:

shym reviewed (approved):
I’ve reviewed the range 8730457 (included)..bbb48274. Very good job!
(The only comment that came to mind is a note to our future selves that it might be interesting to generate `type nonrec` to avoid name collisions. But that’ll require some `module` magic anyway to cover most cases so let’s wait.)
